data/types.go

```diff
@@ -147,3 +160,16 @@ func DefaultExpires(role string) time.Time {
 	}
 	return t.UTC().Round(time.Second)
 }
+
+type unmarshalledSignature Signature
+
+func (s *Signature) UnmarshalJSON(data []byte) error {
+	uSignature := unmarshalledSignature{}
```

Comment:
Wouldn't that cause recursion? Unmarshaling into a different type is what http://attilaolah.eu/2013/11/29/json-decoding-in-go/ suggests.